### PR TITLE
 shell utility to transition alert between cost model to aggregator in 2.4.0

### PR DESCRIPTION
**Pull request by avrodrigues5**

### CODE REVIEW

File: copy-alerts/copy-alerts.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+fi
+
+SOURCE_DIR="/var/configs"
+TARGET_DIR="/var/configs"
+
+# IN cost-model the alerts are stored as alerts.json and in aggregator alerts are stored as alerts-aggregator.json
+OLDFILENAME="alerts/alerts.json"
+NEWFILENAME="alerts/alerts-aggregator.json"
+
+kubectl cp $OLD_POD:"$SOURCE_DIR/$OLDFILENAME" /tmp/"$NEWFILENAME" -c cost-model
```
Comment (ameijer) [inline copy-alerts/copy-alerts.sh:50]: namespace here?

Comment (avrodrigues5) [reply copy-alerts/copy-alerts.sh]: Yup, absolutely needed if were running from other namespace. Fixed it!

File: copy-alerts/copy-alerts.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+set -euxo pipefail
```
Comment (ameijer) [inline copy-alerts/copy-alerts.sh:2]: i'd remove the -x, it really hammers stdout which can confuse folks

File: copy-alerts/copy-alerts.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+# Prompt user for namespace and validate
+read -p "Enter the namespace for your kubecost deployment: " NAMESPACE
```
Comment (ameijer) [inline copy-alerts/copy-alerts.sh:5]: i would instead accept this as an arg. makes it easier to script

File: copy-alerts/copy-alerts.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    echo "Namespace '$NAMESPACE' does not exist. Exiting."
+    exit 1
+fi
+
+OLD_POD=''
+NEW_POD=''
+if [[ $(kubectl -n "$NAMESPACE" get sts -l app=aggregator 2>&1) == *"No resources found"* ]]; then
+    echo "No aggregator StatefulSet found with label 'app=aggregator' in namespace $NAMESPACE"
+    # check if its single cluster deployment
+    if [ $(kubectl -n "$NAMESPACE" get service -l app=aggregator -o name | wc -l) -gt 0 ]; then
```
Comment (ameijer) [inline copy-alerts/copy-alerts.sh:23]: is checking the labels on the service reliable?

File: copy-alerts/copy-alerts.sh [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+fi
+
+# Check if OLD_POD and NEW_POD are not empty
+if [ -z "$OLD_POD" ] || [ -z "$NEW_POD" ]; then
+    echo "Error: OLD_POD or NEW_POD is empty. Exiting."
+    exit 1
+fi
+
+SOURCE_DIR="/var/configs"
+TARGET_DIR="/var/configs"
```
Comment (nik-kc) [inline copy-alerts/copy-alerts.sh:44]: What do you think of making these configurable, as they are in KCM?

File: copy-alerts/copy-alerts.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+# Upload the file to the new reports location in Aggregator
+kubectl cp -n "$NAMESPACE" /tmp/"$NEWFILENAME" $NEW_POD:"$TARGET_DIR/$NEWFILENAME" -c aggregator
+
+# Check the file is transitioned
+output=$(kubectl exec -it -n "$NAMESPACE" $NEW_POD -c aggregator -- ls -lh "$TARGET_DIR/$NEWFILENAME")
+line_count=$(echo "$output" | wc -l)
+
+if [ "$line_count" -eq 1 ]; then
+    echo "File successfully copied and verified:"
```
Comment (nik-kc) [inline copy-alerts/copy-alerts.sh:60]: Is there supposed to be a value after the trailing colon?

File: copy-alerts/copy-alerts.sh [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+else
+    echo "Failed to copy"
```
Comment (nik-kc) [inline copy-alerts/copy-alerts.sh:62]: Could we add more context to this? Why did it fail to copy? Or would that already be logged?

File: copy-alerts/copy-alerts.sh (hunk 4 above)
Comment (avrodrigues5) [reply copy-alerts/copy-alerts.sh:23]: I created a new file called alerts-test-copy.json in aggregator nightly and then updated script to take that as the OLDFILENAME and NEWFILENAME with same suffix and it copied over. 

Screenshot 2024-08-29 at 10 59 10 AM

Screenshot 2024-08-29 at 11 00 11 AM

Screenshot 2024-08-29 at 11 00 06 AM

I think our helm adds label app=aggregator [code-ref](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/templates/aggregator-statefulset.yaml#L51) am i correct here ? 

It's also similar in script in [copy-reports.sh](https://github.com/kubecost/kubecost-utilities/blob/main/copy-reports/copy-reports.sh) . LMK if its enough evidence to answer your question if it's reliable.

File: copy-alerts/copy-alerts.sh (hunk 3 above)
Comment (avrodrigues5) [reply copy-alerts/copy-alerts.sh]: Fixed it.

File: copy-alerts/copy-alerts.sh (hunk 2 above)
Comment (avrodrigues5) [reply copy-alerts/copy-alerts.sh]: removed -x .. Maybe yes but i thought more information is better but agreed too much verbosity.

File: copy-alerts/copy-alerts.sh (hunk 6 above)
Comment (avrodrigues5) [reply copy-alerts/copy-alerts.sh]: removed

File: copy-alerts/copy-alerts.sh (hunk 7 above)
Comment (avrodrigues5) [reply copy-alerts/copy-alerts.sh]: I just check if the file exists in aggregator after the copy. I have enhanced the message LMK if you think there needs more information

File: copy-alerts/copy-alerts.sh (hunk 5 above)
Comment (avrodrigues5) [reply copy-alerts/copy-alerts.sh:50]: I think its better we stick to constant here as we know both go into /var/configs for now unless then have modified it, which i don't think most customers do.  

When you mean configurable a second argument to the script , isnt it?

**Review by ameijer — APPROVED** [review]
LGTM pending @nik-kc 's comments

File: copy-alerts/copy-alerts.sh (hunk 5 above)
Comment (nik-kc) [reply copy-alerts/copy-alerts.sh:50]: Valid point! I think it can remain a constant.

**Review by nik-kc — APPROVED** [review]

**Pull request merged**